File: little-bee/controller/router/area_type.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_AREA_TYPE = "/config/area-type"
const URL_ROUTER_AREA_TYPE_BY_ID = "/config/area-type/:area-type-id"
const URL_ROUTER_AREA_TYPE_QUERY = "/config/area-type/query"
const URL_ROUTER_AREA_TYPE_ALL = "/config/area-type/all"

func AreaTypeRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_AREA_TYPE, config.CreateAreaType)
	r.GET(URL_ROUTER_AREA_TYPE_BY_ID, config.RetrieveAreaType)
	r.PUT(URL_ROUTER_AREA_TYPE, config.UpdateAreaType)
	r.DELETE(URL_ROUTER_AREA_TYPE_BY_ID, config.DeleteAreaType)
	r.PUT(URL_ROUTER_AREA_TYPE_BY_ID, config.DeleteAreaTypeWithFlag)
	r.POST(URL_ROUTER_AREA_TYPE_QUERY, config.QueryAreaType)
	r.GET(URL_ROUTER_AREA_TYPE_ALL, config.GetAllAreaTypes)
	return r
}

File: frontend/console/src/api/area-type.js
import Axios from "@/http";

export function createAreaType(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/area-type', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveAreaType(areaTypeId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/area-type/${ areaTypeId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateAreaType(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/area-type', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteAreaType(areaTypeId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/area-type/${ areaTypeId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteAreaTypeWithFlag(areaTypeId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/area-type/${ areaTypeId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllAreaTypes() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/area-type/all').then(res => {
            resolve(res)
        })
    })
}

export function queryAreaType (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/area-type/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}